blueflood-core/src/main/java/com/rackspacecloud/blueflood/io/AbstractMetricsRW.java

```diff
@@ -148,22 +150,19 @@ protected int getTtl(Locator locator, RollupType rollupType, Granularity granula
      * @param timestampToRollupMap  a map of timestamp to rollup
      * @return
      */
-    protected <T extends Rollup> Points<T> convertToPoints(final Map<Long, T> timestampToRollupMap) {
-        Points<T> rollupPoints =  new Points<T>();
-        for (Map.Entry<Long, T> rollup : timestampToRollupMap.entrySet() ) {
-            rollupPoints.add(new Points.Point<T>(rollup.getKey(), rollup.getValue()));
+    protected <T extends Object> Points<T> convertToPoints(final Map<Long, T> timestampToRollupMap) {
```

Comment:
I am ok with multiple methods. I am ok with ```<T extends Object>``` as well. It is less than ideal, but to do it the ideal way, it is more refactoring that is outside the scope of this PR.